### DISCUSSION TITLE
【Keyball61・左トラックボール】ProMicroが認識されない／ひっくり返すと認識されるが正常に動作しない

### DISCUSSION THREAD

**Discussion by wp041** [Yowkees/keyball#322] [Q&A]
## 概要
Keyball61・左トラックボールでの組み立てにおいて、ビルドガイドを参照しながら組み立てていたところ、手順3-7にて問題が発生しました。

## 詳細
始めに、ビルドガイドの通り、promicroを裏側（Keyball表面から見てpromicro基板の部品が見えない状態）にして組み立てました。その際、左右両方ともPCにpromicroが認識されませんでした。
またこの際、二つともpromicro単体ではPCには認識され、QMKToolboxによるファームウェアの書き込みにも成功しました。そのためpromicro本体は故障していないと考えられます。

これを解消するために、私はpromicro基板の印刷（RAW GND RST ……）とKeyball基板の印刷が左右入れ替わってしまっていることが問題だと判断し、promicroをひっくり返し表側に装着ことで解決させようとしました。
二組目のpromicroを購入し、表側に装着したところ、PCに認識されるものの、ファームウェアの書き込みはできませんでした。
これらのpromicroも単体でファームウェアを書き込むことができたため、ファームウェア書き込み後にKeyballに装着し、PCにつなぎました。結果としてはPCには認識されるがyを連続入力、iuも入力するという挙動を見せました。oledはとても熱くなり、反応しませんでした。
![image](https://github.com/Yowkees/keyball/assets/114465138/6cec4d5f-7748-448e-b0da-8ea3f495df70)

上記のような問題があり、ビルドガイドを先に進めることができません。
どうかご助力頂けないでしょうか。

## 追記
Twitterコミュニティで助けを求めたところ、制作上での問題は見受けられないそうです。
基板の写真を添付します。
![image](https://github.com/Yowkees/keyball/assets/114465138/31610a32-8870-4145-af9f-7a8ff1ae9b9b)

![image](https://github.com/Yowkees/keyball/assets/114465138/8ba925ae-9b50-426f-9916-362a5e246344)

![image](https://github.com/Yowkees/keyball/assets/114465138/95aa3dd9-b471-4c70-b71b-64f0ca132670)

また、[こちら](https://twitter.com/Sonetto_9999/status/1696540176758317412/photo/1)の方のツイートを見ると、私の基盤と印字が違い（左右入れ替え）、これが原因になっていたりするのかなと素人ながらに考えました。

関連リンクとしてtwitter Keyballコミュニティでこの問題についての質問をし、皆さんが協力して下さった際のスレッドのリンクを貼ります。
[twitterスレッドのリンク](https://twitter.com/project_ctrlCV/status/1696362529025642570)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。また、Keyballコミュニティーのみなさんもご助言いただけたようでありがとうございました。

さて、まずProMicroの表裏ですが、ProMicroに部品が実装されている面が下になるのが正解です。左手トラックボール仕様でも、ビルドガイドの写真の通りの向きで取り付けてください。
この向きではProMicroが起動しなかったということでしたが、ProMicro裏の半田付けで隣とショートしている端子がないかご確認ください。テスターをお持ちでしたら、VCCとGNDの刻印がある端子どうしがショートしていないかご確認いただければ幸いです。

お手数ですがご確認お願いいたします。

**Reply by wp041** [reply]
返信どうもありがとうございます。
promicro裏はんだ付けを確認し、ショートしていないことを確認いたしました。
テスターは持っておらず、そのテストはできません……

**Reply by mokesura** [reply]
コミュニティのほうも見ていたのですが、こちらのほうが記録に残りそうでしたのでコメントします。
ProMicroとピンヘッダーをはんだ付けされているとおもいますが、そちらのはんだ付け面も写真を乗せていただくことは可能でしょうか？
ショートしていない、とのことですが、手がかりを見つけられればと思います。

**Reply by wp041** [reply]
返信ありがとうございます。

1組目（色々実験をしたため、破損しています）
![image](https://github.com/Yowkees/keyball/assets/114465138/4a0eb202-e71c-4a98-b5f2-9ab515ae8643)
![image](https://github.com/Yowkees/keyball/assets/114465138/a81808b9-7701-4bbb-b12e-811913489f15)

2組目
![image](https://github.com/Yowkees/keyball/assets/114465138/ec5f0c7e-a548-4bb6-907e-c6555f88e7ec)
![image](https://github.com/Yowkees/keyball/assets/114465138/e5a183c2-3152-47e4-bceb-a6a358402f2d)

このような感じです。
全てで同様の挙動が起きるため、全てがショートしてるとは考えづらく、困っています。

**Reply by mokesura** [reply]
ProMicroの写真の提示ありがとうございます。
はんだ付けには問題なさそうに見えます。
全てのProMicroで同様の挙動が起きているとのことですので、基板側に何かしらの問題がありそうな気がします。
詳しい原因を見るにはテスターが必要になってきそうですが、Keyballユーザーでもないので、詳しいことはヨーキースさん他の方にお願いできればと思います…

**Reply by wp041** [reply]
ご返信ありがとうございます。そうですね、原因があるとするならば基板以外ないのだと思いますが……いかんせん難解で……

**Comment by Yowkees** [comment]
ご助言、誠にありがとうございます。

ProMicroを正しい向き（部品実装面が下）で取り付けたときの挙動はどのようになりましたでしょうか？

**Reply by wp041** [reply]
一切反応をせず、PCにも認識されません。promicro上のLEDは点灯します。

**Comment by Yowkees** [comment]
状況を整理させていただきますと、

ProMicro単体の状態：ProMicroのリセットが出来る。USBデバイスとしてPCに認識される。
ProMicroをミドル基板に挿した状態：ProMicroのリセットができない。USBデバイスとして認識されない。

という状況でよかったでしょうか。
上記２状態の違いは、やはりProMicro裏の端子のどこかがショートしているために起こるものと考えられます。

基板と12Pピンソケットの間に半田が流れ込んでショートすることもありますので、よからぬ端子間がショートしていないか確認が必要だと思われます。

下記にKeyball61の基板データを公開しておりますので、KiCAD等フリーソフトで表示いただければ、GND端子等ショートしていて良い端子と良くない端子が判断いただけると思いますのでご参考にしていただければ幸いです。
https://github.com/Yowkees/keyball/tree/main/keyball61/design_data

以上、お手数ですがご確認お願いいたします。

**Reply by wp041** [reply]
ご返事ありがとうございます。
状況についてその通りです。
いい端子とよくない端子のチェックはテスターを購入し、行えばよろしいのでしょうか？

**Reply by Yowkees** [reply]
ご確認ありがとうございます。
テスターでご確認いただければ幸いです。
テスターがあると今後のキーボードのメンテナンスでも役立つと思いますのでご検討よろしくお願いいたします。

**Reply by wp041** [reply]
購入しました。明日報告いたします。

**Reply by wp041** [reply]
テスターを購入し、ショートの確認を行いました。
12ピンソケットは左右ともにGND-GNDのみのショートで他ショートは一切ありませんでした。
OLED-VCC,OLED-GNDも左右ともに正常で、ショートは一切ありませんでした。
@Yowkees

**Comment by Yowkees** [comment]
ご確認ありがとうございました。

ProMicro側はいかがでしょうか？ピンヘッダーでも、半田がスルーホール内に流れ込み基板とピンヘッダーの間でショートすることがあります。

おっと、ただ、ProMicro側のピンヘッダーがショートしている場合はProMicro単体でも起動しませんね。

ProMicro単体の状態：ProMicroのリセットが出来る。USBデバイスとしてPCに認識される。
ProMicroをミドル基板に挿した状態：ProMicroのリセットができない。USBデバイスとして認識されない。
で、変化ないでしょうか？ピンソケット内でショートが無ければミドル基板に搭載しても起動するはずなのですが。

お手数ですが状況整理のため、再度ご確認いただければ幸いです。

**Reply by wp041** [reply]
ご返信ありがとうございます。

改めてProMicro単体・ミドル基板に装着でPCに認識されるかテストしましたが、やはり
- ProMicro単体の状態：ProMicroのリセットが出来る。USBデバイスとしてPCに認識される。
- ProMicroをミドル基板に挿した状態：ProMicroのリセットができない。USBデバイスとして認識されない。
で症状は変化しませんでした。

また、これは蛇足情報なのかもしれませんが、ProMicroの表裏をひっくり返すと認識されました。

**Comment by Yowkees** [comment]
ご確認ありがとうございます。
状況了解しました。

ProMicroをひっくり返すのは、よからぬ端子同士がショートしてProMicroにダメージがあるかも知れませんので行わないようにお願いします。

今後の調査の進め方についてご相談したいので、下記白銀ラボウェブショップのお問合せよりご連絡いただけますでしょうか。
お手数ですがよろしくお願いいたします。

**Reply by wp041** [reply]
ご返信ありがとうございます。

promicroの件、了解しました。
畏まりました。問い合わせフォームより連絡させていただきます。
